fdbclient/ThreadSafeTransaction.cpp

```diff
@@ -402,10 +402,11 @@ const char* ThreadSafeApi::getClientVersion() {
 }
 
 ThreadFuture<uint64_t> ThreadSafeApi::getServerProtocol(const char* clusterFilePath) {
-	auto [clusterFile, isDefault] = ClusterConnectionFile::lookupClusterFileName(std::string(clusterFilePath));
-
-	Reference<ClusterConnectionFile> f = Reference<ClusterConnectionFile>(new ClusterConnectionFile(clusterFile));
-	return onMainThread([f]() -> Future<uint64_t> { return getCoordinatorProtocols(f); });
+	return onMainThread([clusterFilePath = std::string(clusterFilePath)]() -> Future<uint64_t> {
+		auto [clusterFile, isDefault] = ClusterConnectionFile::lookupClusterFileName(clusterFilePath);
+		Reference<ClusterConnectionFile> f = Reference<ClusterConnectionFile>(new ClusterConnectionFile(clusterFile));
+		return getCoordinatorProtocols(f);
+	});
```

Comment:
Is the race here with the reference counting? I ask because the `ThreadSafeDatabase` constructor is doing something similar. It avoids a potential reference counting issue, but if the problem is something else it might need a look too.

Reply:
Yup, the race here is with reference counting

Reply:
I suppose we should describe what the data race is as part of the PR description in order to meet the requirement of describing the problem.

Reply:
I'll post the tsan output (need to recompile first)